src/components/timeline/TrackVisualProgressGraph.js

```diff
@@ -102,7 +102,7 @@ class TrackVisualProgressCanvas extends React.PureComponent<CanvasProps> {
         // Create a path for the top of the chart. This is the line that will have
         // a stroke applied to it.
         x =
-          (deviceWidth * (progressGraphData[i].timestamp - rangeStart)) /
+          (deviceWidth * ((progressGraphData[i].timestamp ?? 0) - rangeStart)) /
```

Comment:
probably the right way would be to keep the previous value and use it, instead of 0. (and of course use 0 as the initial previous value).
Or maybe we could preprocess it, so that we know that in this component everything is non-null.
 
But this is very theorical for the case where one timestamp is missing. We don't need this at the moment, so I believe this is good enough for now.